addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/LEDStateDTO.java

```diff
@@ -70,18 +75,18 @@ public static LEDStateDTO valueOf(int state, int program, int programSpeed, int
         OnOffType power = (state & 0x01) != 0 ? OnOffType.ON : OnOffType.OFF;
         float[] hsv = new float[3];
         Color.RGBtoHSB(red, green, blue, hsv);
-        long hue = (long) (hsv[0] * 360);
-        int saturation = (int) (hsv[1] * 100);
-        int brightness = (int) (hsv[2] * 100);
+        BigDecimal hue = BigDecimal.valueOf(hsv[0] * 360);
```

Comment:
If it's correct and expected to store the hue here with decimals (I personally do not think that a hue should be so accurate that it stores many decimals and that integer values of 0-255 give enough accuracy to display), please perform a double  calculation and store it as double, or any alternative where storage and calculation are consistent.